src/storage-controller/src/lib.rs

```diff
@@ -1655,6 +1655,46 @@ where
         }
     }
 
+    async fn snapshot_latest(
+        &mut self,
+        id: GlobalId,
+    ) -> Result<Vec<Row>, StorageError<Self::Timestamp>> {
+        let upper = self
+            .persist_monotonic_worker
+            .recent_upper(id)
+            .await
+            .expect("sender hung up")?;
+
+        let res = match upper.as_option() {
+            Some(f) if f > &T::minimum() => {
+                let as_of = f.step_back().unwrap();
+
+                let snapshot = self.snapshot(id, as_of).await.unwrap();
```

Comment:
Do we have to consider the since here? Doesn't it pass the upper in some rare circumstances?

Reply:
It can, but it shouldn't for builtin tables, otherwise they're not readable. This will error with a `StorageError::ReadBeforeSince` when that happens. Which it hopefully doesn't 🤞